pkg/operator/ceph/file/filesystem.go

```diff
@@ -140,6 +141,25 @@ func newFS(name, namespace string) *Filesystem {
 	}
 }
 
+// SetPoolSize method sets the sizes for MetadataPool and dataPool
+func SetPoolSize(context *clusterd.Context, clusterName, fsName string, dataPool []*model.Pool, metadataPool *model.Pool) error {
+	// generating the metadata pool's name
+	metadataPool = generateMetadataPoolName(fsName, metadataPool)
+	err := client.CreatePoolWithProfile(context, clusterName, *metadataPool, appName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update metadata pool %q", *&metadataPool.Name)
+	}
+	// generating the data pool's name
+	dataPool = generateDataPoolNames(fsName, dataPool)
+	for _, pool := range dataPool {
+		err := client.CreatePoolWithProfile(context, clusterName, *pool, appName)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update datapool  %q", *&pool.Name)
```

Comment:
`*&pool.Name` ???

Reply:
ditto

Reply:
At this point `pool` is just a string AFAIK so why that construct?